source/uwp/Renderer/lib/TileControl.cpp

```diff
@@ -0,0 +1,319 @@
+
+namespace AdaptiveNamespace
+{
+    HRESULT TileControl::RuntimeClassInitialize() noexcept try
+    {
+        ComPtr<IContentControlFactory> spFactory;
+        RETURN_IF_FAILED(Windows::Foundation::GetActivationFactory(
+            HStringReference(RuntimeClass_Windows_UI_Xaml_Controls_ContentControl).Get(), &spFactory));
+
+        // Create composable base
+        ComPtr<IInspectable> spInnerInspectable;
+        ComPtr<IContentControl> spInnerContentControl;
+        RETURN_IF_FAILED(spFactory->CreateInstance(static_cast<ITileControl*>(this),
+                                                   spInnerInspectable.GetAddressOf(),
+                                                   spInnerContentControl.GetAddressOf()));
+        RETURN_IF_FAILED(SetComposableBasePointers(spInnerInspectable.Get(), spFactory.Get()));
+
+        // initialize members
+        m_containerElement =
+            XamlHelpers::CreateXamlClass<ICanvas>(HStringReference(RuntimeClass_Windows_UI_Xaml_Controls_Canvas));
+        m_brushXaml = XamlHelpers::CreateXamlClass<IImageBrush>(HStringReference(RuntimeClass_Windows_UI_Xaml_Media_ImageBrush));
+
+        // Add m_containerElement to content of ContentControl
+        ComPtr<IInspectable> content;
+        RETURN_IF_FAILED(m_containerElement.As(&content));
+        RETURN_IF_FAILED(spInnerContentControl->put_Content(content.Get()));
+
+        return S_OK;
+    }
+    CATCH_RETURN;
+
+    _Use_decl_annotations_ HRESULT TileControl::put_BackgroundImage(_In_ IAdaptiveBackgroundImage* value)
+    {
+        m_adaptiveBackgroundImage = value;
+        return S_OK;
+    }
+
+    _Use_decl_annotations_ HRESULT TileControl::put_RootElement(_In_ IFrameworkElement* value)
+    {
+        m_rootElement = value;
+        return S_OK;
+    }
+
+    _Use_decl_annotations_ HRESULT TileControl::get_ResolvedImage(_In_ IUIElement** value)
+    {
+        return m_resolvedImage.CopyTo(value);
+    }
+
+    _Use_decl_annotations_ HRESULT TileControl::put_ImageSize(_In_ ABI::Windows::Foundation::Size value)
+    {
+        m_imageSize = value;
+        return S_OK;
+    }
+
+    _Use_decl_annotations_ HRESULT TileControl::LoadImageBrush(_In_ IUIElement* uielement)
+    {
+        m_resolvedImage = uielement;
+
+        ComPtr<IImage> image;
+        m_resolvedImage.As(&image);
+
+        if (image != nullptr)
+        {
+            ComPtr<IImageSource> imageSource;
+            THROW_IF_FAILED(image->get_Source(&imageSource));
+
+            EventRegistrationToken eventToken;
+            ComPtr<IBitmapImage> bitmapImage;
+            THROW_IF_FAILED(imageSource.As(&bitmapImage));
+            THROW_IF_FAILED(bitmapImage->add_ImageOpened(Callback<IRoutedEventHandler>(
+                                                             [&](IInspectable* /*sender*/, IRoutedEventArgs * /*args*/) -> HRESULT {
+                                                                 ComPtr<IUIElement> uiElement;
```

Comment:
>                                                      [](start = 12, length = 53)

@paulcam206, can you tell what's going on here that makes the auto-formatting so indented? Our other lambdas don't do this, but i'm not sure why this one is different.